content/200-concepts/100-components/02-prisma-client/250-tracing.mdx

```diff
@@ -0,0 +1,362 @@
+  ConsoleSpanExporter,
+  SimpleSpanProcessor,
+} from '@opentelemetry/sdk-trace-base'
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-http'
+import { AsyncHooksContextManager } from '@opentelemetry/context-async-hooks'
+import * as api from '@opentelemetry/api'
+import { PrismaInstrumentation } from '@prisma/instrumentation'
+import { registerInstrumentations } from '@opentelemetry/instrumentation'
+import { trace } from '@opentelemetry/api'
+import { PrismaClient } from '@prisma/client'
+
+/** SETUP */
+
+export function otelSetup() {
+  // a context manager is required to propagate the context
+  const contextManager = new AsyncHooksContextManager().enable()
+
+  // it's for node.js for span nesting and ctx propagation
+  api.context.setGlobalContextManager(contextManager)
+
+  // a simple exporter that logs the raw data to the console
+  const consoleExporter = new ConsoleSpanExporter()
+
+  // exporter that works with Jaeger and other OpenTelemetry compliant collector
+  const otlpTraceExporter = new OTLPTraceExporter()
+
+  // a standard provider that can run on the web and in node
+  const provider = new BasicTracerProvider({
+    // sampler: new TraceIdRatioBasedSampler(0.1),
+    resource: new Resource({
+      // we can define some metadata about the trace resource
+      [SemanticResourceAttributes.SERVICE_NAME]: 'sample-tracing-application',
+      [SemanticResourceAttributes.SERVICE_VERSION]: '1.0.0',
+    }),
+  })
+
+  // send spans to the console
+  provider.addSpanProcessor(new SimpleSpanProcessor(consoleExporter))
+  // send spans to Jaeger or any OpenTelemetry compliant collector
+  provider.addSpanProcessor(new SimpleSpanProcessor(otlpTraceExporter))
+
+  provider.register()
+
+  // Enable tracing in Prisma
+  registerInstrumentations({
+    instrumentations: [new PrismaInstrumentation()],
+  })
+}
+
+//set up tracing for this application
+otelSetup()
+
+// set things up for tracing our "app"
+const tracer = trace.getTracer('Example Application')
+
+async function main() {
+  const prisma = new PrismaClient()
+
+  let users = await tracer.startActiveSpan('create-user', async (span) => {
+    try {
+      const email = `user.${Date.now()}@sample-application.com`
+      await prisma.user.create({
+        data: {
+          email: email,
+        },
+      })
+      let users = await prisma.user.findMany()
+      return users
+    } finally {
+      span.end()
+    }
+  })
+
+  console.table(users)
+}
+
+main().then(() => console.log('example complete'))
+```
+
+## Trace middleware
+
+If you want to include your [middleware](/concepts/components/prisma-client/middleware) in your tracing, then set `middleware` to `true` in your `registerInstrumentations` statement:
+
+```ts
+registerInstrumentations({
+  instrumentations: [new PrismaInstrumentation({ middleware: true })],
+})
+```
+
+## Send tracing to the console
+
+To send tracing to the console, add the following to you application code:
```

Comment:
Isn't there an Otel specific name of what "send" is here? Export? Or does the "console" have a specific Otel name here like "receiver" or something similar? I think Otel has named concepts for all this.

Reply:
Wouldn't it be "Send traces to the console" then? "Tracing" is the overall concept and what we do, what is sent though are the results - the traces. (Yeah, probably nitpicky)

Reply:
Nitpicky is good - getting language right is important! I would say that the user is sending tracing to the console, but I think what I'm using here is a shorthand for "tracing information" or similar. I think traces works as well as this, and is less ambiguous, so I'll change it.

Reply:
Actually, I use the term "send tracing output" elsewhere, and I think I prefer that, so I'll go with that.

Reply:
True, the "tracing output" is in the format of "individual traces" - so this fits.